nxc/protocols/nfs.py

```diff
@@ -273,19 +273,38 @@ def get_file(self):
             self.auth["uid"] = attrs["attributes"]["uid"]
             dir_handle = mnt_info["mountinfo"]["fhandle"]
 
-            # Get the file handle and read the file data
+            # Get the file handle
             dir_data = self.nfs3.lookup(dir_handle, file_name, auth=self.auth)
             file_handle = dir_data["resok"]["object"]["data"]
-            file_data = self.nfs3.read(file_handle, auth=self.auth)
 
-            if "resfail" in file_data:
-                raise Exception("Insufficient Permissions")
-            else:
-                data = file_data["resok"]["data"]
+            # Get the file attributes (size)
+            file_attrs = self.nfs3.getattr(file_handle, auth=self.auth)
+            file_size = file_attrs["attributes"]["size"]
+
+            # Handle files over 32768 bytes
+            offset = 0
+            chunk_size = 32768  # 32KB
```

Comment:
The default chunk size seems to be limited by `1024 * 1024` which is also the default value for the read function.
Can you check if you can for sure only download 32KB at a time? Do you have an idea why that could be? OS specific (which one are you running)?
![image](https://github.com/user-attachments/assets/f0b4e5e9-03f1-4ea4-8278-f08c41826f93)

Reply:
Thanks for getting back to me. I am doing my tests on Kali host, more specifically in an Exegol container in it.

You are right, I modified the code to handle the size with `file_size = dir_data["resok"]["obj_attributes"]["attributes"]["size"]`.

I also implemented the file read with the EOF that is returned from the `read` method.

This is my first PR, so excuse me if I don't know exactly how the process goes. Do I have to make another PR with the changes that I just made or can I add them on to the current one here?

Reply:
> Thanks for getting back to me. I am doing my tests on Kali host, more specifically in an Exegol container in it.
> 
> You are right, I modified the code to handle the size with `file_size = dir_data["resok"]["obj_attributes"]["attributes"]["size"]`.
> 
> I also implemented the file read with the EOF that is returned from the `read` method.
> 
> This is my first PR, so excuse me if I don't know exactly how the process goes. Do I have to make another PR with the changes that I just made or can I add them on to the current one here?

You can add it on your this PR. Thank you btw

Reply:
Ok I think I added it correctly to this PR. Thanks to you for making the NFS protocol with nxc 😉